### DISCUSSION TITLE
Jump list is broken

### DISCUSSION THREAD

**Discussion by momadacoding** [JetBrains/ideavim#415] [General]
I use 1.9.0 eap.1
PyCharm 2021.2.3 (Professional Edition)
Build #PY-212.5457.59, built on October 19, 2021

Runtime version: 11.0.12+7-b1504.40 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1990M
Cores: 6
Registry: ide.balloon.shadow.size=0
Non-Bundled Plugins: com.intellij.ideolog (203.0.27.0), com.samdark.intellij-visual-studio-code-dark-plus (2.5), Key Promoter X (2021.3), com.markskelton.one-dark-theme (5.4.0), com.arcticicestudio.nord.jetbrains (0.13.0), AceJump (3.8.4), IdeaVIM (1.9.0-eap.1), org.jetbrains.IdeaVim-EasyMotion (1.6), io.github.mishkun.ideavimsneak (1.2.0), com.berzanmikaili.intellij.monokai_pro (2.0.2c)

Ctrl-I does not work at all,  
Ctrl-O behavior wrong, it will cycle between last position

**Comment by AlexPl292** [comment]
Hi, could you please also attach your `~/.ideavimrc`?

**Comment by momadacoding** [comment]
""" Map leader to space ---------------------
let mapleader=","

""" Plugins  --------------------------------
set surround
set multiple-cursors
set commentary
set argtextobj
set easymotion
set textobj-entire
set ReplaceWithRegister
set highlightedyank
set which-key
set textobj-indent
set vim-paragraph-motion
set matchit
"set NERDTree

set sneak

""" Plugin settings -------------------------
let g:argtextobj_pairs="[:],(:),<:>"

""" Common settings -------------------------
set showmode
set so=5
set incsearch
set nu
""" use \c or \C to make it case insensitive or sensitive
set ignorecase
set smartcase
""nnoremap * /\<<C-R>=expand('<cword>')<CR>\><CR>
""nnoremap # ?\<<C-R>=expand('<cword>')<CR>\><CR>

""" Idea specific settings ------------------
set ideajoin
set ideastatusicon=gray
set idearefactormode=keep

""" Mappings --------------------------------
map <leader>f <Plug>(easymotion-s)
map <leader>e <Plug>(easymotion-f)

map <leader>d :action Debug<CR>
map <leader>r :action RenameElement<CR>
map <leader>c :action Stop<CR>
map <leader>z :action ToggleDistractionFreeMode<CR>

map <leader>s :action SelectInProjectView<CR>
map <leader>a :action Annotate<CR>
map <leader>h :action Vcs.ShowTabbedFileHistory<CR>
map <leader><space> :action GotoNextError<CR>

map <leader>b :action ToggleLineBreakpoint<CR>
map <leader>o :action FileStructurePopup<CR>

nnoremap <C-]> :action GotoDeclaration<CR>
nnoremap <C-o> :action Back<CR>
nnoremap <C-i> :action Forward<CR>

nmap <silent> <leader>/ :nohl<CR>
" <Tab> is equal <C-I>, so the jumplist is affected
"nmap <Tab> gt
"nmap <S-Tab> gT
nnoremap <silent> <S-t> :tabnew<CR>
" command map
:command WQ wq
:command Wq wq
:command W w
:command Q q

"Avoid the escape key
inoremap jk <ESC>

"" Vmap for maintain Visual Mode after shifting > and <
vmap < <gv
vmap > >gv

"" Split
noremap <Leader>h :<C-u>split<CR>
noremap <Leader>v :<C-u>vsplit<CR>

"U create a new change to undo tree, which contains the last line which was
"modified to its original state, useless
nnoremap U <nop>

noremap YY "+y<CR>
noremap P "+gP<CR>
noremap XX "+x<CR>

vnoremap y myy`y
vnoremap Y myY`y

nnoremap <leader>d "_d
xnoremap <leader>d "_d
xnoremap <leader>p "_dP
vnoremap <leader>p "_dP

"Search Whole word
nnoremap <leader>/ /\<\><left><left>

set clipboard+=unnamed

"set nokeep-english-in-normal-and-restore-in-insert
"set keep-english-in-normal

**Reply by AlexPl292** [reply]
Could you please try to replace 
```
nnoremap <C-o> :action Back<CR>
nnoremap <C-i> :action Forward<CR>
```
With 
```
nmap <C-o> <Action>(Back)
nmap <C-i> <Action>(Forward)
```
?

**Reply by momadacoding** [reply]
It works after replaced

**Reply by momadacoding** [reply]
This works in the latest pycharm, but it broken in older version of pycharm, which I can only install ideavim 0.57

**Reply by AlexPl292** [reply]
That's right, this feature is not implemented in 0.57. What is the reason you have to stay with the older version of pycharm?

**Reply by momadacoding** [reply]
> 

I use pycharm pro, and only has the old version liscense

**Comment by bhootd** [comment]
@AlexPl292 Your suggestion fixed the issue for me too. 

I am wondering about a few things though. Before the fix, my bindings were:

```
nnoremap <C-i> :action Forward<CR>
nnoremap <C-o> :action Back<CR>
```

1. What's the difference between `<Action>(Back)` and `:action Back<CR>`? Which the preferred style? All my actions are defined using `:action Name<CR>`.
2. When I tried `nnoremap <C-o> <Action>(Back)`, the shortcut ends up typing `ck)` a few lines above, clearly misinterpreting the `<Action>(Ba**ck)**`. I am habituated to prefer `{nvio}noremap` to `{nvio}map` in vim in order to avoid recursive bindings. Is the `noremap` versions not preferred in ideavim? In this case, it seems like a recursive scan was required to map `<Action>(Back)` to the actual `Back` action.

**Reply by AlexPl292** [reply]
Hi, 

1) the `<Action>(Back)` notation is a new one and while in general there should be no differences in behaviour between them, it's executed a bit differently and this affects this mapping. So at the moment we suggest to use `<Action>(..)` in mappings, but we keep `:action` to be able to execute actions from the ex line.

2) At the moment, we support only `remap` because it's a kind of more vim-ish. How else would you map to the actual `<Action>(...` characters. However, I see a lot of requests regarding this thing, so probably we'll change this in a way to support `noremap` as well.

**Reply by bhootd** [reply]
> How else would you map to the actual (... characters.

Yes, this crossed my mind while I was typing my post.

I will switch to {nvio}map because its more sensible in ideavim's context: `<C-o>` maps to `<Action>(Back)`, which maps to `Cmd-[`. Its clearly semantically recursive.

I will also switch to `<Action>(name)`.

Thank you for your time!
